Integration Tests/src/Microservice.DataAccessEFCore/Queries/AddNicknames.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microservice.DataAccess.Interfaces;
+using Microservice.DataAccessEFCore.Mappers;
+using Microservice.Domain;
+using Microsoft.EntityFrameworkCore;
+
+namespace Microservice.DataAccessEFCore.Queries
+{
+    public class AddNicknames : IAddNicknames
+    {
+        private readonly NicknamesContext _context;
+
+        public AddNicknames(NicknamesContext context)
+        {
+            _context = context;
+        }
+
+        public async Task ExecuteAsync(string userName, IEnumerable<Nickname> nicknames)
+        {
+            if (userName == null)
+            {
+                throw new ArgumentNullException(nameof(userName));
+            }
+
+            if (nicknames == null)
+            {
+                throw new ArgumentNullException(nameof(nicknames));
+            }
+
+            var user = await _context.Users.FirstOrDefaultAsync(x => x.Name.Equals(userName));
+
+            if (user == null)
+            {
+                throw new Exception();
```

Comment:
Probably that's not the best idea to throw System.Exception. Normally less generic types are recommended.

Reply:
Yes, thanks you noticed this. I added custom exception.